src/panels/config/automation/trigger/types/ha-automation-trigger-time_pattern.ts

```diff
@@ -1,22 +1,22 @@
 import { html, LitElement } from "lit";
 import { customElement, property } from "lit/decorators";
 import { fireEvent } from "../../../../../common/dom/fire_event";
-import type { HaFormSchema } from "../../../../../components/ha-form/types";
+import "../../../../../components/ha-form/ha-form";
 import type { TimePatternTrigger } from "../../../../../data/automation";
 import type { HomeAssistant } from "../../../../../types";
 import type { TriggerElement } from "../ha-automation-trigger-row";
 
-const SCHEMA: HaFormSchema[] = [
+const SCHEMA = [
```

Comment:
I don't like that our `SCHEMA` is no longer typechecked now, I think that might be more important.

Reply:
Could be solved by adding:
```ts
const TYPED_SCHEMA: readonly Readonly<HaFormSchema>[] = SCHEMA;
```
But not really pretty 🤷‍♂️

Reply:
Yeah I agree.  I was thinking the same thing.  Theoretically though, this should be checked in the assignment to the `<ha-form>` property, but that's currently only set to warnings in the plugin:

```
"no-incompatible-type-binding": "warning",
```

And is lit even checked in the workflow?

Reply:
Yeah, it could be checked, but I'm afraid we would have to fix a LOT of other places then.

Reply:
Well, that's not really a bad thing.  Probably worth another branch to see just how bad it would be?  Most places I see that warning are pretty simple fixes involving `null` or `undefined`.  But that's coming from the guy who started this PR, so what do I know. 😇

Reply:
Actually, I just realized that some code I wrote to handle grid schemas might solve the problem:

```typescript
// Type utility to unionize a schema array by flattening any grid schemas
export type SchemaUnion<
  SchemaArray extends readonly HaFormSchema[],
  Schema = SchemaArray[number]
> = Schema extends HaFormGridSchema ? SchemaUnion<Schema["schema"]> : Schema;
```

If you always use this on the schema parameter for the callbacks, then it will be checked against the `extends readonly HaFormSchema[]` constraint, which is the type check you're after.

Reply:
Take a look at the latest commit for examples.  The calendar card schema was not actually being type checked, and the addition of `SchemaUnion` caught an error which I fixed.

Reply:
I would not be against creating a branch with:

`"no-incompatible-type-binding": "error"`

To see if we can fix those issues :-)